freeipa/src/main/java/com/sequenceiq/freeipa/service/multiaz/MultiAzCalculatorService.java

```diff
@@ -188,4 +209,67 @@ private Set<String> collectSubnetIds(InstanceGroup instanceGroup) {
         }
     }
+
+    public void populateAvailabilityZones(Stack stack, DetailedEnvironmentResponse environment, InstanceGroup instanceGroup) {
+        if (!stack.isMultiAz()) {
+            LOGGER.debug("Multi AZ is not enabled for {}", stack.getName());
+            return;
+        }
+        if (instanceGroup.getInstanceGroupNetwork() == null) {
+            instanceGroup.setInstanceGroupNetwork(new InstanceGroupNetwork());
+        }
+        if (!CollectionUtils.isEmpty(instanceGroup.getInstanceGroupNetwork().getAvailabilityZones())) {
+            LOGGER.debug("Availability Zones {} are already provided for Instance Group {} in the request",
+                    instanceGroup.getInstanceGroupNetwork().getAvailabilityZones().stream().collect(Collectors.joining(",")),
+                    instanceGroup.getGroupName());
+            return;
+        }
+        LOGGER.debug("CloudPlatform is {} PlatformVariant is {}", stack.getCloudPlatform(), stack.getPlatformvariant());
+        CloudPlatformVariant cloudPlatformVariant = new CloudPlatformVariant(
+                Platform.platform(stack.getCloudPlatform()),
+                Variant.variant(stack.getPlatformvariant()));
+        AvailabilityZoneConnector availabilityZoneConnector = cloudPlatformConnectors.get(cloudPlatformVariant).availabilityZoneConnector();
+        if (availabilityZoneConnector == null) {
+            LOGGER.debug("Implementation for AvailabilityZoneConnector is not present for CloudPlatform {} and PlatformVariant {}",
+                    stack.getCloudPlatform(), stack.getPlatformvariant());
+            return;
+        }
+        Set<String> availabilityZones = availabilityZoneConnector.getAvailabilityZones(extendedCloudCredentialConverter
+                .convert(credentialService.getCredentialByEnvCrn(stack.getEnvironmentCrn())), environment.getNetwork()
+                        .getAvailabilityZones(CloudPlatform.valueOf(stack.getCloudPlatform())),
+                instanceGroup.getTemplate().getInstanceType(), Region.region(environment.getLocation().getName()));
+        if (availabilityZones.size() < availabilityZoneConnector.getMinZonesForFreeIpa()) {
+            LOGGER.error("Number of zones are less than min number of Zones");
+            throw new BadRequestException(String.format("Based on configured availability zones and instance type, number of available zones " +
+                    "for instance group %s are %s. Please configure at least %s zones for Multi Az deployment", instanceGroup.getGroupName(),
+                    availabilityZones.size(), availabilityZoneConnector.getMinZonesForFreeIpa()));
+        }
+        LOGGER.debug("Availability Zones for instance group are  {}", availabilityZones);
+        instanceGroup.getInstanceGroupNetwork().setAvailabilityZones(availabilityZones);
+    }
+
+    public void populateAvailabilityZonesForInstances(Stack stack, InstanceGroup instanceGroup) {
+        if (!stack.isMultiAz()) {
```

Comment:
I would use if - else if- else statements in this method and remove go to like return statements.

Reply:
Done